### PR TITLE
initial setup

### PR DESCRIPTION
**Pull request by DanielKakona1**
Auth screen and routes added

### CODE REVIEW

File: packages/front/src/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <Container>
+      <p>OwlSell</p>
+      <div
+        style={{
+          backgroundColor: '#1d3557',
+          padding: 10,
+          borderRadius: 3,
+          fontSize: 14
+        }}
```
Comment (brunokiafuka) [inline packages/front/src/components/Header.tsx:14]: Let's avoid inline styles.

File: packages/front/src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      >
+        Kiafuka Store
+      </div>
+    </Container>
+  );
+};
+
+const Container = styled.div`
+  width: 100%;
+  /* max-width: 1366px; */
```
Comment (brunokiafuka) [inline packages/front/src/components/Header.tsx:24]: Do you want to be using styled-components or keep it simple without using `CSS-in-JS`

File: packages/front/src/components/SideBar.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
+// @ts-ignore-file
+import React from 'react';
+import styled from 'styled-components';
+import { NavLink } from 'react-router-dom';
+
+interface Props {
+  routes: any[];
+  history?: any;
+}
```
Comment (brunokiafuka) [inline packages/front/src/components/SideBar.tsx:9]: let's do proper typing 🥶 no `any` should be used unless we do not know the type at all.

File: packages/front/src/components/SideBar.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+  const createLinks = (defaultRoutes: any) =>
+    defaultRoutes.map((prop: any) => {
+      if (prop.layout !== '/dashboard') return null;
+
+      return (
+        <li key={prop.path}>
+          <NavLink to={prop.layout + prop.path}>
+            {/* 
+            // @ts-ignore */}
```
Comment (brunokiafuka) [inline packages/front/src/components/SideBar.tsx:38]: ` @ts-ignore` shouldn't be alloweded.

File: packages/front/src/components/SideBar.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  flex-direction: column;
+  justify-content: space-between;
+  padding: 20px 0;
+  width: 180px;
+  @media (max-width: 650px) {
+    display: none;
+  }
+`;
+
+export const Content = styled.div`
```
Comment (brunokiafuka) [inline packages/front/src/components/SideBar.tsx:74]: You can type the content. `styled.div<T>`

File: packages/front/src/layouts/Dashboard.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import React from 'react';
+import { Route, Switch } from 'react-router-dom';
+import styled from 'styled-components';
+import routes from '../pages';
+import Header from '../components/Header';
+import SideBar from '../components/SideBar';
+import { Container, Sidebar } from 'rsuite';
+
+const Main = styled.main`
+  /*  margin-left: 150px; */
```
Comment (brunokiafuka) [inline packages/front/src/layouts/Dashboard.tsx:11]: comments

File: packages/front/src/layouts/Dashboard.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  width: 100%;
+  @media (max-width: 650px) {
+    margin-left: 0;
+    padding: 0;
+  }
+`;
+
+export default function Dashboard() {
+  const getRoutes = (defaultRoutes: any) =>
+    defaultRoutes.map((prop: any, key: any) => {
```
Comment (brunokiafuka) [inline packages/front/src/layouts/Dashboard.tsx:23]: `any` should be taken care of.

File: packages/front/src/layouts/Dashboard.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            component={prop.component}
+            key={key}
+          />
+        );
+      }
+      return null;
+    });
+
+  return (
+    <div style={{ height: '100vh' }}>
```
Comment (brunokiafuka) [inline packages/front/src/layouts/Dashboard.tsx:41]: iniline styles.

File: packages/front/src/pages/auth/index.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { Container, Content, Form, Button } from './styles'
+import { withRouter } from 'react-router-dom'
+import TextInput from '../../components/TextInput'
+
+interface IProps {
+  history?: any
```
Comment (brunokiafuka) [inline packages/front/src/pages/auth/index.tsx:7]: `any`

File: packages/front/.env [hunk 10]
```diff
@@ -0,0 +1 @@
+SKIP_PREFLIGHT_CHECK=true
```
Comment (brunokiafuka) [inline packages/front/.env:1]: env vars shouldn't be in prod. Please create a `.env.example` file and add the `.env` to the `.gitignore`.  

The `.env.example` file should be where we list the env vars for the app. 

```js
// .env.example
SKIP_PREFLIGHT_CHECK=
...
```

File: packages/front/src/components/Header.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const Header = () => {
+  return (
+    <Container>
+      <p>OwlSell</p>
```
Comment (brunokiafuka) [inline packages/front/src/components/Header.tsx:7]: Change app name to `Eluka`

File: packages/front/src/components/TextInput.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+    registerField({
+      name: fieldName,
+      ref: inputRef.current,
+      path: 'value',
+    })
+  }, [fieldName, registerField])
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', marginBottom: 25 }}>
+      <label style={{ marginBottom: 5, fontSize: 12 }}>{label}</label>
```
Comment (brunokiafuka) [inline packages/front/src/components/TextInput.tsx:33]: inline styles.

File: packages/front/src/layouts/Dashboard.tsx (hunk 7 above)
Comment (DanielKakona1) [reply packages/front/src/layouts/Dashboard.tsx]: I couldn't get the type of props

File: packages/front/.env (hunk 10 above)
Comment (brunokiafuka) [inline packages/front/.env:1]: please push the deletion of this file, so that git can track it.

**Pull request merged**